src/Test.sol

```diff
@@ -139,6 +139,256 @@ abstract contract Test is DSTest {
+            fail();
+        }
+    }
+
+    function assertNotEq(address a, address b, string memory err) internal {
+        if (a == b) {
+            emit log_named_string   ("Error", err);
+            assertNotEq(a, b);
+        }
+    }
+
+    function assertNotEq(bytes32 a, bytes32 b) internal {
+        if (a == b) {
+            emit log                ("Error: a != b not satisfied [bytes32]");
+            emit log_named_bytes32  ("  Expected", b);
+            emit log_named_bytes32  ("    Actual", a);
+            fail();
+        }
+    }
+
+    function assertNotEq(bytes32 a, bytes32 b, string memory err) internal {
+        if (a == b) {
+            emit log_named_string   ("Error", err);
+            assertNotEq(a, b);
+        }
+    }
+
+    function assertNotEq32(bytes32 a, bytes32 b) internal {
+        assertNotEq(a, b);
+    }
+
+    function assertNotEq32(bytes32 a, bytes32 b, string memory err) internal {
+        assertNotEq(a, b, err);
+    }
+
+    function assertNotEq(int a, int b) internal {
+        if (a == b) {
+            emit log            ("Error: a != b not satisfied [int]");
+            emit log_named_int  ("  Expected", b);
+            emit log_named_int  ("    Actual", a);
+            fail();
+        }
+    }
+
+    function assertNotEq(int a, int b, string memory err) internal {
+        if (a == b) {
+            emit log_named_string   ("Error", err);
+            assertNotEq(a, b);
+        }
+    }
+
+    function assertNotEq(uint a, uint b) internal {
+        if (a == b) {
+            emit log            ("Error: a != b not satisfied [uint]");
+            emit log_named_uint ("  Expected", b);
+            emit log_named_uint ("    Actual", a);
+            fail();
+        }
+    }
+
+    function assertNotEq(uint a, uint b, string memory err) internal {
+        if (a == b) {
+            emit log_named_string   ("Error", err);
+            assertNotEq(a, b);
+        }
+    }
+
+    function assertNotEq(string memory a, string memory b) internal {
+        if (keccak256(abi.encodePacked(a)) == keccak256(abi.encodePacked(b))) {
+            emit log                ("Error: a != b not satisfied [string]");
+            emit log_named_string   ("  Value a", a);
+            emit log_named_string   ("  Value b", b);
+            fail();
+        }
+    }
+
+    function assertNotEq(string memory a, string memory b, string memory err) internal {
+        if (keccak256(abi.encodePacked(a)) == keccak256(abi.encodePacked(b))) {
+            emit log_named_string   ("Error", err);
+            assertNotEq(a, b);
+        }
+    }
+
+    /*//////////////////////////////////////////////////////////////////////////
+                                    BOOL ASSERTS
+    //////////////////////////////////////////////////////////////////////////*/
+
+    function assertEq(bool a, bool b) internal virtual {
+        b ? assertTrue(a) : assertFalse(a);
```

Comment:
This will result in a user passing in two values but failing with "Just assertion" failed, instead of differentiating between expected and actual like the rest of the `assertEq` overloads. I'd suggest just changing these to be like the others

```solidity
function assertEq(bool a, bool b) internal {
    if (a != b) {
      emit log("Error: a == b not satisfied [bool]");
      emit log_named_string("  Expected", b ? "true" : "false");
      emit log_named_string("    Actual", a ? "true" : "false");
      fail();
    }
  }
```

Reply:
Fixed by 97f5f77e5ba3c28c2b9440d20202bd954d87da4d

Reply:
New ones lgtm, though I think the old one is still there too—there's two functions with the sig `function assertEq(bool a, bool b, string memory err)`

Reply:
Face palm, fixed. Can resolve if happy